### ISSUE TITLE
Invidious redirect custom options problematic

### ISSUE THREAD

**Issue by frostrenew** [SimonBrazell/privacy-redirect#203]
Currently, there are two issues:

1) URL parameters not always get added to redirected URL, and even if they are, there is currently a regression in Invidious:
https://github.com/iv-org/invidious/issues/1610
that prevents them from working.
2) If one automatically clears cookies and cache after each browser session, the option "persist invidous settings as cookie" has visible effect only on reloading the page.

So, a dumb workaround for the time being could be to auto-refresh redirected URL as soon as a cookie is created, and do it only once per session.

This is probably not a good fix but currently the custom options for Invidious are very unreliable and the sudden change from light to dark mode on going to another Invidious URL really takes away from the experience.

**Comment by moriel5** [comment]
I was about to create a new issue for this, however I see it already exists, so this makes things more organized.

I personally have cache and cookies cleared after every session, however for me, the settings aren't applied at all, even after refreshes, and even if cookies and cache aren't cleared at session close.

**Comment by SimonBrazell** [comment]
@frostrenew 

I'm glad to see someone making use of the "persist invidous settings as cookie" 🙂 I've only had people complain about it requiring the use of cookies so far...

There is a content script that does this for a select list of invidious instances, so perhaps it can be tweaked to improve it's behaviour, and like you suggest at a bare minimum it could refresh the page when it sets the cookie.

@moriel5 

The content script that sets the cookie only loads on a predetermined list of invidious instances, these are currently:

        "*://invidious.snopyta.org/*",
        "*://invidious.xyz/*",
        "*://invidious.kavin.rocks/*",
        "*://tube.connect.cafe/*",
        "*://invidious.zapashcanon.fr/*",
        "*://invidiou.site/*",
        "*://vid.mint.lgbt/*",
        "*://invidious.site/*",
        "*://yewtu.be/*",
        "*://invidious.tube/*",
        "*://invidious.silkky.cloud/*",
        "*://invidious.fdn.fr/*",
        "*://invidious.himiko.cloud/*",
        "*://inv.skyn3t.in/*",
        "*://tube.incognet.io/*",
        "*://invidious.tinfoil-hat.net/*",
        "*://invidious.namazso.eu/*",
        "*://vid.puffyan.us/*",
        "*://dev.viewtube.io/*",
        "*://invidious.048596.xyz/*",
        "*://fz253lmuao3strwbfbmx46yu7acac2jz27iwtorgmbqlkurlclmancad.onion/*",
        "*://qklhadlycap4cnod.onion/*",
        "*://c7hqkpkpemu6e7emz5b4vyz7idjgdvgaaa3dyimmeojqbgpea3xqjoid.onion/*",
        "*://w6ijuptxiku4xpnnaetxvnkc5vqcdu7mgns2u77qefoixi63vbvnpnqd.onion/*"

Are you perhaps using an instance that isn't included in this list?

**Comment by moriel5** [comment]
@SimonBrazell Note, I am currently using Namazso's instance, and it wasn't working for me with Snopyta's instance.

- [ ] Regarding the feature, I have it enabled, since when I had started using Privacy Redirect, my preferences only got applied that way (mainly for enabling DASH and disabling video proxying).

I do have cookies and cache set to be cleared at session close, however keeping them between sessions doesn't change the situation.

**Comment by SimonBrazell** [comment]
> I am currently using Namazso's instance

@moriel5 looks like that instance was only added in the latest release, are you running v1.1.47? And if not can you try it out and see if you are still having the same problem? Should be live for Firefox, not for Chromium or Edge yet though as their review process takes a few days.

**Comment by moriel5** [comment]
Is 1.1.47 the version that came out on Friday?
If so, then yes, the issue is still there (I use both Firefox Nightly (main) and Edge Dev (secondary), and at the time I was on Firefox).

**Comment by SimonBrazell** [comment]
@moriel5 yeah that's it, I'll try your setup with my machine and see if I have the same experience, let me know what instance you are using (copy from the extension settings) and which Invidious settings you are trying to persist and I'll try it out.

**Comment by moriel5** [comment]
Currently Instance:
https://invidious.namazso.eu

Settings I want to persist:
Always proxy videos to through Invidious: False
Invidious Video Quality: DASH (I would also like an option for selecting what DASH resultion I want, as that is an Invidious option)
Invidious Dark Mode always On: False
Invidious Player Style: YouTube
Invidious automatically play video on load: False
